### ISSUE TITLE
Winforms style designer for Avalonia?

### ISSUE THREAD

**Issue by Shadowblitz16** [AvaloniaUI/Avalonia#4361]
Can we get a xaml designer similar to winform's?
I think we can all agree that visual studio's xaml designer is pretty poor unless your writing xaml and know what your doing.

I think something similar to winforms in which you can position and resize your controls and see the rectangle clearly.
and something like the winforms property browser but with bindable options for each property.

the reason why I am making this suggestion is I hate xaml with a passion. 
now I know most people don't and thats why a lot of gui frameworks are in xaml but I would like a way of not having to touch it that doesn't involve using the default xaml designer

**Comment by kekekeks** [comment]
>which you can position and resize your controls and see the rectangle clearly.

That's UI building approach is outdated. Like, two decades outdated. If we are going to get a UI designer, it will work more like GTK's Glade where you can drag controls into predefined layout slots.

![glade](https://user-images.githubusercontent.com/1067584/88462865-4595be80-ceb7-11ea-9bcf-62dfde3630ab.gif)

**Comment by danwalmsley** [comment]
@Shadowblitz16 its quite a difficult task to acheive that, I hope that something like what @kekekeks has suggested will be possible at some point in the future, but it would take someone dedicated to developing and maintaining that.

**Comment by Shadowblitz16** [comment]
@kekekeks I like that one too.
I do think that each property should have a circle icon where you can bind things to it though

**Comment by prayank23** [comment]
I am surprised there is no designer for AXAML like XAML designer in Visual Studio. It would save lot of time. Couldn't find anything in Visual Studio, Avalon Studio, Rider etc. and different questions on stackoverflow and reddit.

Is there is any way to convert XAML created using XAML designer in Visual Studio to AXAML? A tool that automatically changes everything based on what is supported in Avalonia.

**Comment by amwx** [comment]
> 
> 
> I am surprised there is no designer for AXAML like XAML designer in Visual Studio. It would save lot of time. Couldn't find anything in Visual Studio, Avalon Studio, Rider etc. and different questions on stackoverflow and reddit.
> 
> Is there is any way to convert XAML created using XAML designer in Visual Studio to AXAML? A tool that automatically changes everything based on what is supported in Avalonia.

.axaml is .xaml. The axaml file extension was added because Visual Studio kept altering the .csproj file to basically remove regular xaml files when they were added to the project, which created a headache b/c you had to go remove those additions in order to build the project. 
The designer Avalonia projects use is a custom editor in the VS extension b/c I believe the default WPF/UWP xaml designers don't work well with other frameworks (or are incompatible?). The bigger problem is that 99% of things regarding building extensions and designers in VS is undocumented (or outdated) so nothing makes sense and it's all just sort of hacked together, thus it's very minimal in features. It also doesn't help the VS team broke it even more with a change to intellisense a few versions back.

**Comment by maxkatz6** [comment]
@amwx you are right, but I believe @prayank23 was asking about converting WPF xaml to Avalonia XAML, which is not possible.

**Comment by blackholeearth** [comment]
i whip up with this in 5 hour.  in winforms net4.5 ..

assume that im using  ava's  canvas  absolute positioning .

[avaNetApp_TargetDesktop.zip](https://github.com/user-attachments/files/17952876/avaNetApp_TargetDesktop.zip)

![Animation](https://github.com/user-attachments/assets/70ba7998-5596-4b16-9bb6-eda9ef099291)

but this is  winforms exe. i dont know how to make  extensions. looks tons of trouble.
 
but probably cant do the property  explorer section... that is currently beyond me, as i have never worked wit it.

-----

somebody tell me..  can i do this.  inside avalonia?   

psuedo code;;

```
Form1_load()
{ 
     var xamlSTR = File.ReadAllText("myAvaView1.axaml");

     var  myAvaView1 =  avalonia.loadXAML(xamlSTR );

          myAvaView1.applyThemes() //?? theme  from app or str??
          myAvaView1.render_or_Show() ;   
           
           //now its on screen .  user will drag and move them?? 
          // this is availabe in winforms   you just set property "allow user to move resize in runtime"
}

```

 if this is possible ,  

 
 i im noob at ava .. i cound even get datagrid work as i did in winforms..

**Comment by Shadowblitz16** [comment]
I still need this.

Something like winforms form builder for rider would be nice but I also need tolling for theming my ui.
Avalonia is too complex to understand how to make my own ui themes.

I hear avalonia uses styling similar to css.
If this is the case having support for selectors and animations in the designer would also be good.

Winforms designer is not outdated, its just using an windows only ui.
If winforms was a wrappers around win32, cocao, gtk and qt and was better at theming and databinding it would still be used today.

**Comment by theLaffa** [comment]
If it helps anyone stumbling over this issue, I've found https://xaml.io/ has been able to get me most of the way. It's not really designed for Avalonia but does well enough at least for placing components and sorting out layouts.

**Comment by AbdalaMask** [comment]
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/30e0df68-ebc7-49cf-803b-413104945c14" />WpfDesigner To Avalonia
